### PR TITLE
Create R53 SRV records for newly deployed cluster

### LINKED ISSUE #29: Contact points discovery
Probably using DNS records.

### PR DESCRIPTION
**Pull request by a1exsh**
Closes #29

### CODE REVIEW

File: create_cluster.py [hunk 1]
```diff
@@ -239,6 +239,34 @@ def get_subnets(prefix_filter: str, regions: list) -> dict:
...
+
+    for region, ips in node_ips.items():
+        with Action('Setting up Route53 SRV records in {}..'.format(region)):
+            r53.change_resource_record_sets(
+                HostedZoneId=zone['Id'],
+                ChangeBatch={
+                    'Changes': [{
+                        'Action': 'UPSERT',
+                        'ResourceRecordSet': {
+                            'Name': '_{}-{}._tcp.{}'.format(cluster_name, region, hosted_zone),
```
Comment (hjacobs) [inline create_cluster.py]: do we validate the `cluster_name` somewhere (regex)? we need to make sure that it works here as a DNS name

Comment (hjacobs) [inline create_cluster.py]: I would add a regex validator for the `cluster_name` click argument (proposed regex: `[a-z][a-z0-9-][a-z0-9]`)

Comment (a1exsh) [inline create_cluster.py]: I would let Route53 complain if it doesn't like anything (by the way, it seems that Security Groups naming rules are more strict than of Route53).  But I can also see value in having a simple regexp check.

**Comment by a1exsh** [comment]
Actually, we want to record Private IPs in the DNS even when we go multi-region, because it's supposed to be queried by the clients, not Cassandra itself (the configuration still lists public IPs of the seed nodes).

**Comment by a1exsh** [comment]
Tested and updated, please review. :-)

File: create_cluster.py [hunk 2]
```diff
@@ -157,54 +157,57 @@ def generate_certificate(cluster_name: str):
...
-    for region in regions:
-        with Action('Allocating Public IPs for {}..'.format(region)) as act:
+def allocate_ip_addresses(region_subnets: dict, cluster_size: int, node_ips: dict,
+                          take_elastic_ips: bool):
+    '''
+    Allocate unused private IP addresses by checking the current
+    reservations, and optionally allocate Elastic IPs.
+    '''
+    for region, subnets in region_subnets.items():
+        with Action('Allocating IP addresses in {}..'.format(region)) as act:
```
Comment (slitsche) [inline create_cluster.py]: The act.progress() call seems to be gone. Could it be removed?

**Comment by a1exsh** [comment]
And I've just found out (the hard way) that recording IP addresses instead of hostnames was a bad idea: they come out of DNS lookup with an added dot at the end, which totally breaks the Cassandra driver...  Will push a fix for that.

**Comment by Jan-M** [comment]
:+1:

**Pull request merged**